src/Security/Authorization.cs

```diff
@@ -172,23 +172,27 @@ public static Authorization Create (AuthorizationParameters parameters, Authoriz
 				unsafe {
 					if (parameters != null){
 						ppars = &pars;
-						pars.ptrToAuthorization = (AuthorizationItem *) Marshal.AllocHGlobal (sizeof (AuthorizationItem) * 3);
+						pars.ptrToAuthorization = (AuthorizationItem *) Marshal.AllocHGlobal (sizeof (AuthorizationItem) * 2);
 						if (parameters.PathToSystemPrivilegeTool != null)
 							EncodeString (ref pars.ptrToAuthorization [pars.count++], "system.privilege.admin", parameters.PathToSystemPrivilegeTool);
-						if (parameters.Prompt != null)
-							EncodeString (ref pars.ptrToAuthorization [pars.count++], "prompt", parameters.Prompt);
 						if (parameters.IconPath != null)
 							EncodeString (ref pars.ptrToAuthorization [pars.count++], "prompt", parameters.IconPath);
```

Comment:
not related to the PR but using `"prompt"` here seems incorrect, c.c. @chamons

Reply:
Yeah I opened a separate issue for the icon/prompt problem #3613

I tried to fix the icon path but could not prove it was working.